scripts/fetchdep.sh

```diff
@@ -22,15 +22,18 @@ clone() {
 }
 
 # Try the PR author's branch in case it exists on the deps as well.
-# First we check if BUILDKITE_BRANCH is defined,
+# First we check if GITHUB_HEAD_REF is defined,
```

Comment:
Great, thanks for adding this! Does it also work for PRs from forked repos as well? Historically, we've had to make extra changes to support forks, but perhaps that's already covered...?

Reply:
As far as I know the branch matching with forked is already covered a little bit below in the script where it creates the `BRANCH_ARRAY` variable

Reply:
Ah okay, I wasn't sure what `GITHUB_HEAD_REF` would contain for a fork, but anyway if it doesn't work, we can always improve again later.

Reply:
You were right, BuildKite slightly differently than GitHub Actions. The latter does not embed the author username in the branch reference
This is now fixed and is working, you can see the log of a fork working https://github.com/matrix-org/matrix-react-sdk/runs/2876602650?check_suite_focus=true#step:4:28